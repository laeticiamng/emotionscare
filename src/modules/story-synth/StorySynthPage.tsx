"use client";
import React from "react";
import { PageHeader, Card, Button, AudioPlayer } from "@/COMPONENTS.reg"; // AudioPlayer (P5) si dispo
import { synthParagraphs } from "@/lib/story-synth/templates";
import { addStory, loadStories } from "@/lib/story-synth/store";
import { downloadText } from "@/lib/story-synth/export";
import { ff } from "@/lib/flags/ff";
import { recordEvent } from "@/lib/scores/events"; // P6 (toléré si absent)

type Genre = "calme"|"aventure"|"poetique"|"mysterieux"|"romance";
type Pov = "je"|"il"|"elle"|"nous";
type Style = "sobre"|"lyrique"|"journal"|"dialogue";

const AMBIENTS: Record<string, string> = {
  doux: "/audio/lofi-120.mp3",
  pluie: "/audio/rain-soft.mp3"
};

export default function StorySynthPage() {
  const [genre, setGenre] = React.useState<Genre>("calme");
  const [pov, setPov] = React.useState<Pov>("je");
  const [hero, setHero] = React.useState("Alex");
  const [place, setPlace] = React.useState("la ville");
  const [length, setLength] = React.useState(4);
  const [style, setStyle] = React.useState<Style>("sobre");
  const [seed, setSeed] = React.useState("");
  const [ambient, setAmbient] = React.useState<keyof typeof AMBIENTS | "aucun">("aucun");

  const [title, setTitle] = React.useState("Petite histoire");
  const [story, setStory] = React.useState<string[]>([]);
  const [lib, setLib] = React.useState(() => loadStories());

  const newAudio = ff?.("new-audio-engine") ?? false;

  function onGenerate() {
    const paragraphs = synthParagraphs({ genre, pov, hero, place, length, style, seed });
    setStory(paragraphs);
    try {
      recordEvent?.({
        module: "story-synth",
        startedAt: new Date().toISOString(),
        endedAt: new Date().toISOString(),
        durationSec: 10 + length * 3,
        score: Math.min(10, 2 + length),
        meta: { genre, pov, style }
      });
    } catch {}
  }

  function onContinue() {
    const more = synthParagraphs({ genre, pov, hero, place, length: Math.min(2, 2), style, seed: seed || String(Date.now()) });
    setStory(s => [...s, ...more.slice(0, 2)]);
  }

  function onSave() {
    if (!story.length) return;
    const rec = {
      id: crypto.randomUUID?.() || String(Date.now()),
      createdAt: new Date().toISOString(),
      title: title.trim() || "Histoire",
      content: story,
      tags: [genre, style],
      mood: genre
    };
    addStory(rec);
    setLib(loadStories());
    try {
      recordEvent?.({
        module: "story-synth",
        startedAt: rec.createdAt,
        endedAt: new Date().toISOString(),
        durationSec: Math.max(15, story.join(" ").length / 20),
        score: Math.min(10, Math.round(story.length / 2)),
        meta: { saved: true }
      });
    } catch {}
  }

  function onExport() {
    if (!story.length) return;
    downloadText(`${(title || "histoire").replace(/\s+/g,"-").toLowerCase()}.txt`, story);
  }

  return (
    <main aria-label="Story Synth">
      <PageHeader title="Story Synth" subtitle="Génère une courte histoire à partir d’un brief" />
      <Card>
        <section style={{ display:"grid", gap: 10 }}>
          <div style={{ display:"grid", gap: 8 }}>
            <label>Genre
              <select value={genre} onChange={(e)=>setGenre(e.target.value as Genre)}>
                <option value="calme">Calme</option>
                <option value="aventure">Aventure</option>
                <option value="poetique">Poétique</option>
                <option value="mysterieux">Mystérieux</option>
                <option value="romance">Romance</option>
              </select>
            </label>

            <label>Point de vue
              <select value={pov} onChange={(e)=>setPov(e.target.value as Pov)}>
                <option value="je">Je</option>
                <option value="il">Il</option>
                <option value="elle">Elle</option>
                <option value="nous">Nous</option>
              </select>
            </label>

            <label>Protagoniste
              <input type="text" value={hero} onChange={(e)=>setHero(e.target.value)} />
            </label>

            <label>Lieu
              <input type="text" value={place} onChange={(e)=>setPlace(e.target.value)} />
            </label>

            <label>Longueur : {length} §
              <input type="range" min={3} max={7} value={length} onChange={(e)=>setLength(parseInt(e.target.value,10))} />
            </label>

            <label>Style
              <select value={style} onChange={(e)=>setStyle(e.target.value as Style)}>
                <option value="sobre">Sobre</option>
                <option value="lyrique">Lyrique</option>
                <option value="journal">Journal</option>
                <option value="dialogue">Dialogue</option>
              </select>
            </label>

            <label>Seed (optionnel)
              <input type="text" value={seed} onChange={(e)=>setSeed(e.target.value)} placeholder="Reproductible si renseigné"/>
            </label>

            <label>Ambiance sonore
              <select value={ambient} onChange={(e)=>setAmbient(e.target.value as any)}>
                <option value="aucun">Aucun</option>
                <option value="doux">Lofi doux</option>
                <option value="pluie">Pluie</option>
              </select>
            </label>

            {newAudio && ambient !== "aucun" && (
              <AudioPlayer
                src={AMBIENTS[ambient]}
                title={`Ambiance ${ambient}`}
                loop
                defaultVolume={0.5}
              />
            )}
          </div>

          <div style={{ display:"flex", gap: 8, flexWrap:"wrap" }}>
            <Button onClick={onGenerate} data-ui="primary-cta">Générer</Button>
            <Button onClick={onContinue}>Continuer (+)</Button>
            <Button onClick={onSave}>Enregistrer</Button>
            <Button onClick={onExport}>Exporter .txt</Button>
          </div>

          <label>Titre
            <input type="text" value={title} onChange={(e)=>setTitle(e.target.value)} />
          </label>

          {!!story.length && (
            <article style={{ display:"grid", gap: 10 }}>
              {story.map((p, i) => <p key={i} style={{ whiteSpace: "pre-wrap" }}>{p}</p>)}
            </article>
          )}
        </section>
      </Card>

      <Card style={{ marginTop: 12 }}>
        <h2>Bibliothèque (local)</h2>
        <ul style={{ listStyle:"none", padding:0, display:"grid", gap:8 }}>
          {lib.map(s => (
            <li key={s.id} style={{ border:"1px solid var(--card)", borderRadius:12, padding:10 }}>
              <div style={{ display:"flex", justifyContent:"space-between", gap:8 }}>
                <strong>{s.title}</strong>
                <small>{new Date(s.createdAt).toLocaleString()}</small>
              </div>
              <p style={{ marginTop:6, opacity:.85 }}>{(s.content[0] || "").slice(0,120)}…</p>
            </li>
          ))}
          {!lib.length && <em>Aucune histoire enregistrée pour l’instant.</em>}
        </ul>
      </Card>
    </main>
  );
}
