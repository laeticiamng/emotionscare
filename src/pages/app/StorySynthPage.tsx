import React, { useState, useRef, useEffect } from 'react';
import { Play, Pause, RotateCcw, Volume2, VolumeX, BookOpen, Sparkles } from 'lucide-react';
import { Button } from '@/components/ui/button';
import { Card } from '@/components/ui/card';
import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from '@/components/ui/select';
import { Switch } from '@/components/ui/switch';
import { toast } from 'sonner';

interface Story {
  id: string;
  title: string;
  content: string;
  theme: string;
  duration: number;
  audioUrl?: string;
}

const StorySynthPage = () => {
  const [currentStory, setCurrentStory] = useState<Story | null>(null);
  const [selectedTheme, setSelectedTheme] = useState('forest');
  const [isGenerating, setIsGenerating] = useState(false);
  const [isPlaying, setIsPlaying] = useState(false);
  const [showSubtitles, setShowSubtitles] = useState(true);
  const [currentSentence, setCurrentSentence] = useState(0);
  const [isOffline, setIsOffline] = useState(false);
  const audioRef = useRef<HTMLAudioElement>(null);

  const themes = {
    forest: {
      label: 'For√™t Enchant√©e',
      description: 'Promenades paisibles sous les arbres',
      icon: 'üå≤',
      color: 'from-green-500 to-emerald-500'
    },
    ocean: {
      label: 'Oc√©an Infini',
      description: 'Vagues douces et horizons apaisants',
      icon: 'üåä',
      color: 'from-blue-500 to-cyan-500'
    },
    sky: {
      label: 'Ciel √âtoil√©',
      description: 'Contemplation nocturne et s√©r√©nit√©',
      icon: '‚≠ê',
      color: 'from-purple-500 to-indigo-500'
    },
    garden: {
      label: 'Jardin Secret',
      description: 'D√©couverte d\'un havre de paix',
      icon: 'üå∏',
      color: 'from-pink-500 to-rose-500'
    }
  };

  // Histoires pr√©-g√©n√©r√©es par th√®me
  const sampleStories: Record<string, Story> = {
    forest: {
      id: 'forest-1',
      title: 'La Clairi√®re Dor√©e',
      content: 'Vous marchez sur un sentier moussu, entour√© de grands ch√™nes centenaires. Les rayons du soleil filtrent doucement √† travers les feuilles, cr√©ant une danse de lumi√®re dor√©e sur le sol. Un ruisseau murmure tout pr√®s, accompagnant vos pas d\'une m√©lodie cristalline. Vous arrivez dans une clairi√®re circulaire o√π l\'herbe tendre vous invite √† vous reposer. Ici, le temps s\'arr√™te. Vous respirez profond√©ment l\'air pur et frais, sentant toute tension quitter votre corps. Les oiseaux chantent une berceuse apaisante tandis qu\'une brise l√©g√®re caresse votre visage.',
      theme: 'forest',
      duration: 180
    },
    ocean: {
      id: 'ocean-1', 
      title: 'La Plage √âternelle',
      content: 'Vos pieds nus touchent le sable fin et chaud d\'une plage d√©serte. Devant vous, l\'oc√©an s\'√©tend √† l\'infini, d\'un bleu profond et hypnotisant. Les vagues arrivent en rythme r√©gulier, se brisant doucement sur le rivage dans un murmure apaisant. Vous vous asseyez face √† l\'horizon, laissant le soleil r√©chauffer votre peau. Chaque vague qui se retire emporte avec elle vos soucis et vos tensions. Le sel dans l\'air purifie vos pens√©es. Vous fermez les yeux et vous laissez bercer par cette symphonie naturelle, sentant votre esprit s\'all√©ger comme une plume dans la brise marine.',
      theme: 'ocean',
      duration: 170
    },
    sky: {
      id: 'sky-1',
      title: 'La Nuit des √âtoiles',
      content: 'Allong√© sur une couverture douce, vous contemplez la vo√ªte c√©leste. Des milliers d\'√©toiles scintillent au-dessus de vous, formant des constellations myst√©rieuses. La lune, bienveillante, baigne le paysage d\'une lumi√®re argent√©e apaisante. L\'air nocturne est frais et pur, portant les parfums subtils de la nuit. Vous sentez la vastitude de l\'univers vous envelopper d\'un sentiment de paix profonde. Chaque √©toile semble vous murmurer que vous √™tes exactement l√† o√π vous devez √™tre. Le silence cosmique berce votre √¢me et vous vous sentez connect√© √† quelque chose de plus grand, de plus sage.',
      theme: 'sky',
      duration: 165
    },
    garden: {
      id: 'garden-1',
      title: 'Le Refuge Fleuri',
      content: 'Vous poussez une petite porte de bois et d√©couvrez un jardin secret, cach√© du monde ext√©rieur. Des roses parfum√©es grimpent le long des murs de pierre ancienne, m√™lant leurs couleurs tendres. Un petit bassin accueille des n√©nuphars qui flottent paisiblement. Le parfum des lavandes et du jasmin embaume l\'air ti√®de. Vous vous installez sur un banc de marbre sous une pergola couverte de glycines violettes. Ici, dans ce sanctuaire naturel, votre c≈ìur trouve son rythme apais√©. Les papillons dansent de fleur en fleur, vous rappelant la beaut√© simple des petits bonheurs. Chaque respiration vous remplit de s√©r√©nit√©.',
      theme: 'garden',
      duration: 175
    }
  };

  // V√©rifier le statut offline
  useEffect(() => {
    const updateOnlineStatus = () => setIsOffline(!navigator.onLine);
    updateOnlineStatus();
    
    window.addEventListener('online', updateOnlineStatus);
    window.addEventListener('offline', updateOnlineStatus);
    
    return () => {
      window.removeEventListener('online', updateOnlineStatus);
      window.removeEventListener('offline', updateOnlineStatus);
    };
  }, []);

  const generateStory = async () => {
    setIsGenerating(true);
    
    try {
      // Simulation de g√©n√©ration
      await new Promise(resolve => setTimeout(resolve, 2000));
      
      // Utiliser l'histoire pr√©-g√©n√©r√©e pour le th√®me
      const story = sampleStories[selectedTheme];
      if (story) {
        setCurrentStory(story);
        setCurrentSentence(0);
        
        if (isOffline) {
          toast.success('Histoire g√©n√©r√©e (mode hors ligne)', {
            description: 'Lecture texte disponible'
          });
        } else {
          toast.success('Histoire g√©n√©r√©e avec succ√®s', {
            description: 'Pr√™te pour la lecture audio'
          });
        }
      }
    } catch (error) {
      toast.error('Erreur de g√©n√©ration', {
        description: 'Utilisez une histoire pr√©-charg√©e'
      });
    } finally {
      setIsGenerating(false);
    }
  };

  const playStory = async () => {
    if (!currentStory) return;

    if (isOffline) {
      // Mode texte uniquement
      setIsPlaying(true);
      startTextReading();
      toast.info('Mode hors ligne', {
        description: 'Lecture texte activ√©e avec sous-titres'
      });
    } else {
      // Simulation lecture audio
      setIsPlaying(true);
      startTextReading();
      toast.success('Lecture audio d√©marr√©e', {
        description: 'Installez-vous confortablement'
      });
    }
  };

  const pauseStory = () => {
    setIsPlaying(false);
  };

  const resetStory = () => {
    setIsPlaying(false);
    setCurrentSentence(0);
    setCurrentStory(null);
    toast.info('Histoire termin√©e', {
      description: 'Que voulez-vous faire maintenant ?'
    });
  };

  const startTextReading = () => {
    if (!currentStory) return;
    
    const sentences = currentStory.content.split('. ');
    let sentenceIndex = 0;
    
    const readNextSentence = () => {
      if (sentenceIndex < sentences.length && isPlaying) {
        setCurrentSentence(sentenceIndex);
        sentenceIndex++;
        
        setTimeout(readNextSentence, 3000); // 3 secondes par phrase
      } else if (sentenceIndex >= sentences.length) {
        setIsPlaying(false);
        toast.success('Histoire termin√©e !', {
          description: 'Prenez quelques instants pour savourer ce moment de calme'
        });
      }
    };
    
    readNextSentence();
  };

  const formatDuration = (seconds: number) => {
    const mins = Math.floor(seconds / 60);
    const secs = seconds % 60;
    return `${mins}:${secs.toString().padStart(2, '0')}`;
  };

  const getCurrentText = () => {
    if (!currentStory) return '';
    const sentences = currentStory.content.split('. ');
    return sentences[currentSentence] || '';
  };

  return (
    <div className="min-h-screen bg-gradient-to-br from-slate-900 via-blue-900 to-slate-900 p-4">
      <div className="max-w-4xl mx-auto">
        {/* Header */}
        <div className="text-center mb-8">
          <h1 className="text-4xl font-bold bg-gradient-to-r from-blue-400 to-purple-400 bg-clip-text text-transparent mb-2">
            Story Synth
          </h1>
          <p className="text-slate-300">
            Histoires apaisantes g√©n√©r√©es pour votre d√©tente
          </p>
          {isOffline && (
            <p className="text-amber-400 text-sm mt-2">
              Mode hors ligne - Lecture texte disponible
            </p>
          )}
        </div>

        {currentStory ? (
          /* Lecteur d'histoire */
          <div className="space-y-6">
            <Card className="p-8 bg-slate-800/30 backdrop-blur-sm border-slate-700">
              <div className="text-center space-y-6">
                <div>
                  <div className="flex items-center justify-center gap-2 mb-4">
                    <span className="text-3xl">{themes[currentStory.theme as keyof typeof themes]?.icon}</span>
                    <h2 className="text-2xl font-bold text-white">{currentStory.title}</h2>
                  </div>
                  <p className="text-slate-400">
                    Dur√©e estim√©e : {formatDuration(currentStory.duration)}
                  </p>
                </div>

                {/* Zone de sous-titres */}
                {showSubtitles && isPlaying && (
                  <div className="bg-slate-900/50 p-6 rounded-lg border border-slate-600">
                    <p className="text-lg text-slate-200 leading-relaxed">
                      {getCurrentText()}
                    </p>
                  </div>
                )}

                {/* Texte complet si pas en lecture */}
                {(!isPlaying || !showSubtitles) && (
                  <div className="bg-slate-900/30 p-6 rounded-lg border border-slate-700 max-h-64 overflow-y-auto">
                    <p className="text-slate-300 leading-relaxed text-left">
                      {currentStory.content}
                    </p>
                  </div>
                )}

                {/* Contr√¥les */}
                <div className="flex justify-center gap-4">
                  {!isPlaying ? (
                    <Button onClick={playStory} size="lg" className="bg-blue-600 hover:bg-blue-700">
                      <Play className="w-5 h-5 mr-2" />
                      {isOffline ? 'Lire' : '√âcouter'}
                    </Button>
                  ) : (
                    <Button onClick={pauseStory} variant="secondary" size="lg">
                      <Pause className="w-5 h-5 mr-2" />
                      Pause
                    </Button>
                  )}
                  
                  <Button onClick={resetStory} variant="outline" size="lg">
                    <RotateCcw className="w-5 h-5 mr-2" />
                    Nouvelle histoire
                  </Button>
                </div>
              </div>
            </Card>

            {/* Options */}
            <Card className="p-6 bg-slate-800/30 backdrop-blur-sm border-slate-700">
              <div className="flex items-center justify-between">
                <div className="flex items-center gap-2">
                  {showSubtitles ? <Volume2 className="w-5 h-5 text-blue-400" /> : <VolumeX className="w-5 h-5 text-slate-500" />}
                  <label className="text-white font-medium">Sous-titres</label>
                </div>
                <Switch 
                  checked={showSubtitles}
                  onCheckedChange={setShowSubtitles}
                />
              </div>
            </Card>
          </div>
        ) : (
          /* S√©lection de th√®me */
          <div className="space-y-6">
            <Card className="p-6 bg-slate-800/30 backdrop-blur-sm border-slate-700">
              <div className="space-y-4">
                <h3 className="text-lg font-semibold text-white flex items-center gap-2">
                  <BookOpen className="w-5 h-5" />
                  Choisissez votre univers
                </h3>
                
                <Select value={selectedTheme} onValueChange={setSelectedTheme}>
                  <SelectTrigger className="bg-slate-700 border-slate-600 text-white">
                    <SelectValue />
                  </SelectTrigger>
                  <SelectContent>
                    {Object.entries(themes).map(([key, theme]) => (
                      <SelectItem key={key} value={key}>
                        <div className="flex items-center gap-2">
                          <span>{theme.icon}</span>
                          <span>{theme.label}</span>
                        </div>
                      </SelectItem>
                    ))}
                  </SelectContent>
                </Select>

                <Button 
                  onClick={generateStory}
                  disabled={isGenerating}
                  className="w-full bg-gradient-to-r from-blue-600 to-purple-600 hover:from-blue-700 hover:to-purple-700"
                >
                  {isGenerating ? (
                    <>
                      <Sparkles className="w-5 h-5 mr-2 animate-spin" />
                      Cr√©ation en cours...
                    </>
                  ) : (
                    <>
                      <Sparkles className="w-5 h-5 mr-2" />
                      G√©n√©rer l'histoire
                    </>
                  )}
                </Button>
              </div>
            </Card>

            {/* Aper√ßu du th√®me s√©lectionn√© */}
            <div className="grid md:grid-cols-2 gap-4">
              {Object.entries(themes).map(([key, theme]) => (
                <Card 
                  key={key} 
                  className={`p-6 cursor-pointer transition-all duration-200 ${
                    selectedTheme === key 
                      ? 'bg-slate-700/50 border-blue-500' 
                      : 'bg-slate-800/30 border-slate-700 hover:border-slate-600'
                  }`}
                  onClick={() => setSelectedTheme(key)}
                >
                  <div className="text-center space-y-3">
                    <div className={`w-16 h-16 mx-auto bg-gradient-to-r ${theme.color} rounded-full flex items-center justify-center text-2xl`}>
                      {theme.icon}
                    </div>
                    <div>
                      <h4 className="font-semibold text-white">{theme.label}</h4>
                      <p className="text-sm text-slate-400">{theme.description}</p>
                    </div>
                  </div>
                </Card>
              ))}
            </div>

            {/* Informations */}
            <Card className="p-6 bg-slate-800/30 backdrop-blur-sm border-slate-700">
              <h3 className="font-semibold text-white mb-4">Comment √ßa fonctionne</h3>
              <ul className="text-sm text-slate-300 space-y-2">
                <li>‚Ä¢ Histoires apaisantes de 3 √† 5 minutes</li>
                <li>‚Ä¢ Lecture audio avec sous-titres optionnels</li>
                <li>‚Ä¢ Mode hors ligne disponible (texte uniquement)</li>
                <li>‚Ä¢ Th√®mes vari√©s pour tous les go√ªts</li>
                <li>‚Ä¢ Con√ßues pour favoriser la relaxation</li>
              </ul>
            </Card>
          </div>
        )}
      </div>
    </div>
  );
};

export default StorySynthPage;