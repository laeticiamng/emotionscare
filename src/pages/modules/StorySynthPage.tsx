import React, { useState, useCallback } from 'react';
import { motion } from 'framer-motion';
import { BookOpen, Moon, Heart, Zap, Play, Pause } from 'lucide-react';
import { UniverseEngine } from '@/components/universe/UniverseEngine';
import { RewardSystem } from '@/components/rewards/RewardSystem';
import { useRewardsStore } from '@/store/rewards.store';
import { useOptimizedAnimation } from '@/hooks/useOptimizedAnimation';
import { universes } from '@/data/universes/config';
import { Button } from '@/components/ui/button';

interface Story {
  id: string;
  title: string;
  theme: 'calm' | 'confidence' | 'sleep' | 'energy';
  icon: React.ElementType;
  duration: number;
  description: string;
  content: string[];
}

const StorySynthPage: React.FC = () => {
  const [isEntering, setIsEntering] = useState(true);
  const [phase, setPhase] = useState<'setup' | 'selection' | 'listening' | 'complete'>('setup');
  const [selectedStory, setSelectedStory] = useState<Story | null>(null);
  const [currentSegment, setCurrentSegment] = useState(0);
  const [isPlaying, setIsPlaying] = useState(false);
  const [showReward, setShowReward] = useState(false);
  
  const { addReward } = useRewardsStore();
  const { entranceVariants, cssAnimationClasses } = useOptimizedAnimation();

  const universe = universes.storySynth || {
    name: "La Bibliothèque Vivante",
    ambiance: {
      colors: { 
        background: "linear-gradient(135deg, #1a1625 0%, #2d1b3d 50%, #4a2c5a 100%)",
        primary: "#9c88ff",
        secondary: "#b794f6" 
      },
      metaphor: "Chaque histoire est un chemin vers ton mieux-être"
    }
  };

  const stories: Story[] = [
    {
      id: 'calm-ocean',
      title: 'L\'Océan Éternel',
      theme: 'calm',
      icon: Heart,
      duration: 5,
      description: 'Une histoire pour apaiser le mental',
      content: [
        "Tu te trouves sur une plage de sable fin, les vagues caressent doucement le rivage...",
        "Le bruit de l'océan devient une mélodie apaisante qui ralentit ton rythme cardiaque...",
        "Chaque respiration synchronise avec le mouvement des vagues...",
        "Tu ressens une paix profonde s'installer en toi...",
        "Cette sérénité t'accompagnera bien au-delà de ce moment."
      ]
    },
    {
      id: 'mountain-strength',
      title: 'La Montagne Intérieure',
      theme: 'confidence',
      icon: Zap,
      duration: 4,
      description: 'Découvre ta force intérieure',
      content: [
        "Tu gravits un sentier de montagne, chaque pas te rend plus fort...",
        "La vue devient plus claire à mesure que tu montes...",
        "Tu réalises que cette force était en toi depuis toujours...",
        "Au sommet, tu vois tout ce dont tu es capable."
      ]
    },
    {
      id: 'starlight-sleep',
      title: 'La Lumière des Étoiles',
      theme: 'sleep',
      icon: Moon,
      duration: 6,
      description: 'Un voyage vers un sommeil réparateur',
      content: [
        "Les étoiles brillent doucement au-dessus de toi...",
        "Chaque étoile emporte une tension de ta journée...",
        "Ton corps devient de plus en plus lourd et détendu...",
        "Les soucis s'évaporent dans la nuit cosmique...",
        "Tu dérivs vers un sommeil profond et réparateur...",
        "Demain sera un nouveau jour plein de possibilités."
      ]
    }
  ];

  const handleEnterComplete = useCallback(() => {
    setIsEntering(false);
    setPhase('selection');
  }, []);

  const selectStory = useCallback((story: Story) => {
    setSelectedStory(story);
    setPhase('listening');
    setCurrentSegment(0);
  }, []);

  const togglePlayback = useCallback(() => {
    setIsPlaying(!isPlaying);
  }, [isPlaying]);

  const nextSegment = useCallback(() => {
    if (!selectedStory) return;
    
    if (currentSegment < selectedStory.content.length - 1) {
      setCurrentSegment(prev => prev + 1);
    } else {
      completeStory();
    }
  }, [selectedStory, currentSegment]);

  const completeStory = useCallback(() => {
    if (!selectedStory) return;
    
    setPhase('complete');
    setIsPlaying(false);
    
    const reward = addReward({
      type: 'sticker',
      name: 'Phrase-Poème',
      description: `Souvenir de "${selectedStory.title}"`
    });

    setTimeout(() => {
      setShowReward(true);
    }, 800);
  }, [selectedStory, addReward]);

  // Auto-advance segments when playing
  React.useEffect(() => {
    if (!isPlaying || !selectedStory) return;

    const timer = setTimeout(() => {
      nextSegment();
    }, 8000); // 8 seconds per segment

    return () => clearTimeout(timer);
  }, [isPlaying, currentSegment, selectedStory, nextSegment]);

  if (phase === 'setup' || phase === 'selection') {
    return (
      <UniverseEngine
        universe={universe}
        isEntering={isEntering}
        onEnterComplete={handleEnterComplete}
        enableParticles={true}
        className="bg-gradient-to-br from-indigo-900 via-purple-900 to-pink-900"
      >
        <div className="min-h-screen flex flex-col items-center justify-center p-6">
          <motion.div
            className="text-center mb-12"
            variants={entranceVariants}
            initial="hidden"
            animate={phase === 'selection' ? "visible" : "hidden"}
          >
            <div className="inline-flex items-center gap-3 mb-4">
              <BookOpen className="w-8 h-8 text-purple-400" />
              <h1 className="text-3xl font-light text-white">La Bibliothèque Vivante</h1>
            </div>
            <p className="text-white/70 text-lg max-w-md">
              Chaque histoire est un chemin vers ton mieux-être
            </p>
          </motion.div>

          {phase === 'selection' && (
            <div className="grid gap-4 max-w-md w-full">
              <h3 className="text-white/80 text-center mb-4">Choisis ton voyage</h3>
              {stories.map((story, index) => (
                <motion.div
                  key={story.id}
                  initial={{ opacity: 0, x: -20 }}
                  animate={{ opacity: 1, x: 0 }}
                  transition={{ delay: index * 0.15 }}
                >
                  <Button
                    onClick={() => selectStory(story)}
                    className="w-full bg-gradient-to-r from-purple-500/20 to-pink-500/20 hover:from-purple-500/30 hover:to-pink-500/30 text-white rounded-2xl p-6 border border-purple-400/30"
                  >
                    <story.icon className="w-6 h-6 mr-3 text-purple-400" />
                    <div className="text-left flex-1">
                      <p className="font-medium">{story.title}</p>
                      <p className="text-sm text-white/60">{story.description}</p>
                      <p className="text-xs text-white/40">{story.duration} min</p>
                    </div>
                  </Button>
                </motion.div>
              ))}
            </div>
          )}
        </div>
      </UniverseEngine>
    );
  }

  if (phase === 'listening') {
    const progress = selectedStory ? (currentSegment / selectedStory.content.length) * 100 : 0;
    
    return (
      <UniverseEngine
        universe={universe}
        isEntering={false}
        onEnterComplete={() => {}}
        enableParticles={true}
        className="bg-gradient-to-br from-indigo-900 via-purple-900 to-pink-900"
      >
        <div className="min-h-screen flex flex-col items-center justify-center p-6">
          <motion.div
            className="text-center mb-8"
            initial={{ opacity: 0, y: 20 }}
            animate={{ opacity: 1, y: 0 }}
          >
            <h2 className="text-2xl font-light text-white mb-2">
              {selectedStory?.title}
            </h2>
            <div className="w-64 bg-white/20 rounded-full h-2 mb-4">
              <div 
                className="bg-gradient-to-r from-purple-400 to-pink-400 h-2 rounded-full transition-all duration-1000"
                style={{ width: `${progress}%` }}
              />
            </div>
            <p className="text-white/60 text-sm">
              Segment {currentSegment + 1} sur {selectedStory?.content.length}
            </p>
          </motion.div>

          {/* Story Content */}
          <motion.div
            className="bg-white/10 backdrop-blur-md rounded-3xl p-8 max-w-lg w-full mb-8"
            key={currentSegment}
            initial={{ opacity: 0, y: 20 }}
            animate={{ opacity: 1, y: 0 }}
            transition={{ duration: 0.5 }}
          >
            <p className="text-white text-lg leading-relaxed text-center">
              {selectedStory?.content[currentSegment]}
            </p>
          </motion.div>

          {/* Controls */}
          <div className="flex gap-4">
            <Button
              onClick={togglePlayback}
              className="w-16 h-16 rounded-full bg-gradient-to-r from-purple-500 to-pink-500 hover:from-purple-600 hover:to-pink-600 text-white"
            >
              {isPlaying ? <Pause className="w-6 h-6" /> : <Play className="w-6 h-6" />}
            </Button>
            
            <Button
              onClick={nextSegment}
              className="px-6 py-3 bg-white/20 hover:bg-white/30 text-white rounded-full"
            >
              Suivant
            </Button>
          </div>
        </div>
      </UniverseEngine>
    );
  }

  return (
    <UniverseEngine
      universe={universe}
      isEntering={false}
      onEnterComplete={() => {}}
      enableParticles={true}
      className="bg-gradient-to-br from-indigo-900 via-purple-900 to-pink-900"
    >
      <div className="min-h-screen flex flex-col items-center justify-center p-6">
        <motion.div
          className="text-center mb-8"
          initial={{ opacity: 0, y: 20 }}
          animate={{ opacity: 1, y: 0 }}
        >
          <div className="inline-flex items-center gap-3 mb-4">
            <div className={`w-12 h-12 bg-purple-400 rounded-full flex items-center justify-center ${cssAnimationClasses.pulse}`}>
              <BookOpen className="w-6 h-6 text-white" />
            </div>
            <h2 className="text-2xl font-light text-white">Histoire vécue</h2>
          </div>
          <p className="text-white/70">
            Emporte cette sérénité avec toi
          </p>
        </motion.div>

        {/* Poetic Quote */}
        <motion.div
          className="bg-white/10 backdrop-blur-md rounded-3xl p-6 max-w-md w-full mb-8"
          initial={{ opacity: 0, scale: 0.9 }}
          animate={{ opacity: 1, scale: 1 }}
          transition={{ delay: 0.5 }}
        >
          <p className="text-white text-center italic font-light">
            "Dans chaque histoire se cache une vérité qui nous aide à grandir"
          </p>
        </motion.div>

        {showReward && (
          <RewardSystem
            type="sticker"
            message="Belle histoire vécue"
            onComplete={() => setShowReward(false)}
          />
        )}
      </div>
    </UniverseEngine>
  );
};

export default StorySynthPage;