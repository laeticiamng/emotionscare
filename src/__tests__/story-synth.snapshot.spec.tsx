import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import StorySynthPage from "@/modules/story-synth/StorySynthPage";

describe("StorySynthPage", () => {
  it("rend la page et le CTA Générer", () => {
    const { container, getByText } = render(<StorySynthPage />);
    expect(getByText(/Story Synth/i)).toBeTruthy();
    expect(getByText(/Générer/i)).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
