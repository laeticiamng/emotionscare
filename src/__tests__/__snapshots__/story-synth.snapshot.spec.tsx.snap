// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StorySynth Page > rend la page 1`] = `
<div>
  <main
    class="p-4 space-y-4"
  >
    <div>
      StorySynth
    </div>
    <p>
      Bienvenue dans le module StorySynth.
    </p>
    <button>
      Commencer
    </button>
  </main>
</div>
`;
